src/models/network.js

```diff
@@ -63,14 +74,27 @@ Network.prototype.setNick = function(nick) {
 	);
 };
 
-Network.prototype.toJSON = function() {
-	return _.omit(this, [
-		"awayMessage",
-		"chanCache",
-		"highlightRegex",
-		"irc",
-		"password",
-	]);
+/**
+ * Get a clean clone of this network that will be sent to the client.
+ * This function performs manual cloning of network object for
+ * better control of performance and memory usage.
+ *
+ * Both of the parameters that are accepted by this function are passed into channels' getFilteredClone call.
+ *
+ * @see {@link Chan#getFilteredClone}
+ */
+Network.prototype.getFilteredClone = function(lastActiveChannel, lastMessage) {
+	return Object.keys(this).reduce((newNetwork, prop) => {
+		if (prop === "channels") {
+			// Channels objects perform their own cloning
+			newNetwork[prop] = this[prop].map((channel) => channel.getFilteredClone(lastActiveChannel, lastMessage));
```

Comment:
It seems super weird that `lastActiveChannel` can be either an int or a boolean (essentially bypassing the int). Wouldn't make more sense to do this at the caller's level, something like `channel.getFilteredClone(channel.id === lastActiveChannel, lastMessage)`?

Reply:
This is effectively emulating overloaded methods since JS doesn't have these.

Reply:
Meh, I really don't like this. This is not a construct the language has, it's odd to try to emulate it and can lead to bugs. But oh well. Glad this is merged either way.